e2/src/main/index.ts

```diff
@@ -144,10 +150,21 @@ async function whenReady() {
 // for applications and their menu bar to stay active until the user quits
 // explicitly with Cmd + Q.
 app.on('window-all-closed', () => {
-  if (process.platform !== 'darwin') {
-    app.quit()
+  if (process.platform == 'darwin') {
+    updateAndRestartNowIfNeeded();
+  } else {
+    app.quit();
   }
-})
+});
+
+async function updateAndRestartNowIfNeeded() {
+  if (update?.updateInfo.version) {
```

Comment:
```suggestion
  if (await update?.downloadPromise) {
```

Maybe we should check if the update is downloaded before installing?

Reply:
```suggestion
  await update?.downloadPromise;
  if (update?.updateInfo.version) {
```